app/main.py

```
@@ -1,25 +1,76 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: IntegerRange, name: str) -> None:
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: IntegerRange, owner: IntegerRange) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: IntegerRange, value: int) -> None:
+        if (self.min_amount <= value <= self.max_amount):
+            setattr(instance, self.protected_name, True)
+        else:
+            setattr(instance, self.protected_name, False)
```

No, in this case you will not know actual value, in case not in range you should raise an exception, otherwise set correct value